crates/orchestrator/src/workers/mod.rs

```diff
@@ -1,13 +1,39 @@
-use std::error::Error;
-
+use crate::{config::config, jobs::types::JobStatus};
 use async_trait::async_trait;
+use std::error::Error;
 
+pub mod data_submission;
 pub mod proof_registration;
 pub mod proving;
 pub mod snos;
 pub mod update_state;
 
 #[async_trait]
 pub trait Worker: Send + Sync {
+    async fn run_worker_if_enabled(&self) -> Result<(), Box<dyn Error>> {
```

Comment:
interesting, do you see a use case to have a separate function for this?

Reply:
Allows for generalised guard clause checks, no need to rewrite same code on every worker.